source/architecture.rst

```diff
@@ -1,11 +1,89 @@
-Architecture of S-ENDA
-"""""""""""""""""""""""
+Architecture of S-ENDA Find
+"""""""""""""""""""""""""""
 
-..
+S-ENDA architecture is described using the C4 model (https://c4model.com/)
+C4 does not define any properties based on the directionality
+of the used arrows so each arrow should have a textual
+description to avoid disambiguity.
 
-   S-ENDA architecture is described using the C4 model (https://c4model.com/)
+We draft the initial version of the architecture with emphasis
+on the goal and use cases of the first sprint of the project.
+
+The goal
+========
+To draft the architecture for S-ENDA find and access,
+and establish metadata handling to lift the user experience
+in finding dynamic geodata to a new level
+
+
+The use cases
+=============
+- An outdoor swimming competition organizer
+  wants to know sea-water temperature at the event location #37
+- A user shall extract observed and forecasted temperature
+  time series data values over Longyearbyen #35
+- Farmer in Malawi use case #2
+- En bruker ved miljødirektoratet vil undersøke
+  vegetasjonstilstand i et gitt område basert på målinger fra Sentinel-2 #31
+- Miljødirektoratet skal bruke dynamiske geodata
+  data i sitt beslutningssystem #16
+- A tourist guide wants to look at the cloud for a specific
+  region at night for the visibility of Northern lights #38
 
 Context diagram
-================
+===============
+
+The figure below illustrates the top level view of the architecture.
+A mock of "S-ENDA Find" is the product of Sprint 1. The system interfaces
+with data and service providers via a Web UI and a console app. The
+"S-ENDA Find" system exposes a CSW interface for external portals, so that
+they can harvest metadata information about datasets previously registered
+with "S-ENDA Find". "S-ENDA Find" also interacts with the DataCite API when
+a DOI needs to be created or updated.
 
    .. uml:: context.puml
+
+
+Users
+-----
+
+The goal of the project is to make sure that all kinds of potential users
+of dynamical geodata will be able to find and use the data. There is a broad
+spectrum of users with varying expertise in data handling and domain knowledge
+when it comes to dynamical geodata.
+
+General user
+############
+Any user interested in dynamical geodata.
+
+Advanced user
+#############
+An experienced user who knows how to access and process data in their tool
+of choice. Besides the WebUI portals they need a machine-to-machine interface,
+so that they can integrate with their code or command line tools.
+
+Providers
+---------
+
+Data provider
```

Comment:
The distinction between services and data is unclear here. As far as I understood data was about specific datasets from, e.g., an instrument, whereas services target specific users who are only interested in specific geophysical parameters, e.g., air temperature (from any source). Am I correct?

Reply:
However - if this definition is correct; does it matter then? Also an OPeNDAP access protocol would be a (data) service, right? Why do we need to split them when both are regarding data access?

I propose to add the change this paragraph to the following:

# Service providers

Producers of dynamical geophysical datasets that should be made discoverable and accessible for users.

Reply:
Can we stick to "dynamic geodata"? That is defined in the Data Management Handbook as:
> Data describing geophysical processes which are continuously evolving over time. Typically 
> these data are used for monitoring and prediction of the weather, sea, climate and environment.

"geophysical data" is also used in the DMH, but unfortunately not defined.

When it comes to services I see them as instances of fex. "thredds", "a WMS server", "an api serving timeseries in xml or json", then service providers are those developing and operating these services.

Reply:
yes, dynamic(al?) geodata is fine. I interpret it as the same, but it sounds better and is shorter :)

Thredds, wms, timeseries, xml, json, opendap - to my understanding these are just different flavors of data services, aren't they? 

There is a difference between *providing*, *developing* and *operating*, though. But here we are talking about *providing*, right?

Reply:
> yes, dynamic(al?) geodata is fine. I interpret it as the same, but it sounds better and is shorter :)

Ok, then we stick with that. I also think it's a broader concept as that gives space to other dynamic geodata, such as marine biology, which might not fit the geophysical set.

> Thredds, wms, timeseries, xml, json, opendap - to my understanding these are just different flavors of data services, aren't they?

Yes, also we often tend to name our services after a particular software or protocol (just like I've done now :), which complicates things.

> There is a difference between _providing_, _developing_ and _operating_, though. But here we are talking about _providing_, right?

Agree. And maybe even just a part of that. We talk about "registering" metadata about a service that is being provided.

Reply:
> Ok, then we stick with that. I also think it's a broader concept as that gives space to other dynamic geodata, such as marine biology, which might not fit the geophysical set.

Good point!

> > Thredds, wms, timeseries, xml, json, opendap - to my understanding these are just different flavors of data services, aren't they?
> > There is a difference between _providing_, _developing_ and _operating_, though. But here we are talking about _providing_, right?
> 
> Agree. And maybe even just a part of that. We talk about "registering" metadata about a service that is being provided.

Ok - but if we agree that we're talking about **data services** - isn't it enough with one box? Why do we need two (*data provider* and *service provider*)?

Reply:
I am confused by this: Are "service provider" and "data provider" people that registers a service and a dataset?

If so, then we have the corresponding "data service" and "dataset" as separate things right?

Reply:
> Ok - but if we agree that we're talking about **data services** - isn't it enough with one box? Why do we need two (_data provider_ and _service provider_)?

Mostly because I am thinking of these as different entities. At least how we have it today, most data providers don't operate their storage and distribution services

Reply:
> I am confused by this: Are "service provider" and "data provider" people that registers a service and a dataset?
> 
> If so, then we have the corresponding "data service" and "dataset" as separate things right?

Yes, I think **data service** and **dataset** should eventually appear somewhere.

Reply:
I think that `dataset` and `data service` are separate things in our metadata, but I don't think its necessary to start with having separate `Container` for registration. 

I suggested calling the API for registration of metadata `Geodynamic assets API` to signal that this one service handles registration/editing/validation of both services and datasets.  And in the `Find` system both `dataset` and `data service` are just metadata. When we come to the `Access`system, there will be separate `Container` for each data service, and one or more datastores that contains the actual  datasets.

Reply:
> I'm not sure about this... I feel we could use one box called "Data Service Provider". The operations is not really an issue here is it? Isn't this more about the actual registration of "data services"? How does operation come in? Even if this process was made automatic, there would be some kind of dataset registration, right? Do we need to add another box covering operation?

A service needs to be maintained (operated) so that users can access the datasets that they have been able to find. Now I see that _accessing_ has been crossed out at the whiteboard when it comes to Sprint 1, so I suggest to ignore these aspects and concentrate on _Finding_.

Reply:
> I think that `dataset` and `data service` are separate things in our metadata, but I don't think its necessary to start with having separate `Container` for registration.

Agree. That is how I drew it now. We have a single registration API and a single metadata store. There are two (or three, if one counts the API itself) user-facing frontends used to register the _assets_, i.e., dynamic geodata and services related to them.
 
> I suggested calling the API for registration of metadata `Geodynamic assets API` to signal that this one service handles registration/editing/validation of both services and datasets. And in the `Find` system both `dataset` and `data service` are just metadata. When we come to the `Access`system, there will be separate `Container` for each data service, and one or more datastores that contains the actual datasets.

**Geodynamics**
> is a subfield of geophysics dealing with dynamics of the Earth. It applies physics, chemistry and mathematics to the understanding of how mantle convection leads to plate tectonics and geologic phenomena such as seafloor spreading, mountain building, volcanoes, earthquakes, faulting and so on. It also attempts to probe the internal activity by measuring magnetic fields, gravity, and seismic waves, as well as the mineralogy of rocks and their isotopic composition. Methods of geodynamics are also applied to exploration of other planets.

We should not re-use the word in a new context unless completely necessary. The existing words would be either **dynamic geodata (and services)** (from the Data Management Handbook) or **geospatiotemporal (data/services)** .

Reply:
I think I understand the "service provider" now, and have suggested some changes in the text. Please have a look to see if you agree.

Apart from that, I think it is important to stick to **dynamic geodata (and services)**, as suggested by @mpejcoch 

Regarding the name of the API; there is no definition of an *action* between the API container and the metadata store. I suggested *ingestion of metadata* in a comment. Is this correct? At least, I think it would be easier to agree about a name if we have a clear idea of what the actions are...